plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -582,7 +582,7 @@ def set_pre_stop
         # Default termination grace period for pods
         (container[:lifecycle] ||= {})[:preStop] ||= {exec: {command: ["/bin/sleep", pre_stop_sleep.to_s]}}
       end
-      if pre_stop_sleep.to_i > DEFAULT_TERMINATION_GRACE_PERIOD
+      if pre_stop_sleep.to_i >= DEFAULT_TERMINATION_GRACE_PERIOD
```

Comment:
technically if it's 29 we should still set to 32 to maintain the diff

```suggestion
      extra = 3
      if pre_stop_sleep.to_i + extra >= DEFAULT_TERMINATION_GRACE_PERIOD
        pod_template[:spec][:terminationGracePeriodSeconds] = pre_stop_sleep + extra
```